src/object/authenticate.c

```diff
@@ -4107,7 +3592,12 @@ update_cache (MOP classop, SM_CLASS * sm_class, AU_CLASS_CACHE * cache)
 
   need_pop_er_stack = true;
 
-  bits = &cache->data[Au_cache_index];
+  bits = get_cache_bits (sm_class);
```

Comment:
By this update, the third argument,  AU_CLASS_CACHE * cache, of update_cache() can be removed because it is unused in the function.
Note that he two call sites of update_cache(), one from check_authorization() and the other from au_get_class_privilege(), sends a uninitialized garbage value to this cache argument.

Reply:
Thank you. I've removed the redundant arg at 6820282.